### DISCUSSION TITLE
schema driven check?

### DISCUSSION THREAD

**Discussion by pm100** [dotenv-linter/dotenv-linter#758] [Ideas]
I was surprised to see that there isnt already this option. 

Define a schema that says what values are permitted / required/ banned. And what values they can take.

**Comment by pm100** [comment]
I have a rough working version that uses a schema like this

```
{
    "version": "1.0.0",
    "allow_other_keys": true,
    "entries": [
        {
            "key": "SECRET",
            "required": true,
            "allowed_type": "String"
        },
        {
            "key": "PORT",
            "required": true,
            "allowed_type": "Number"
        },
        {
            "key": "URL",
            "required": true,
            "allowed_type": "Url"
        },
        {
            "key": "EMAIL",
            "required": true,
            "allowed_type": "Email"
        }
    ]
}
```

I can add other features such as
- allowed numeric range
- allowed values for string or number
- ..

The 'required' attribute requires a change to the checker calling structure, the checker needs to be called at the end so it cam report missing values. I have not done that.

**Reply by DDtKey** [reply]
I like the concept!
It's definitely should be extended to support possible options. 

And we can extend it step by step: range for numbers, regex for string(?) and etc


But the interesting thing that we invent our own schema-specification for .env files. Need to think how to evolve it and manage versions. Separate subcrate for parsing + spec should be available at least 🤔

**Comment by pm100** [comment]
pr #761
